util/test/sub_test.py

```diff
@@ -2425,29 +2426,34 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
-                    sys.stdout.flush()
-                    p = py3_compat.Popen([utildir+'/test/computePerfStats',
-                                         perfexecname, perfdir, keyfile, execlog, str(exectimeout), perfdate],
-                                         stdout=subprocess.PIPE)
-                    sys.stdout.write('%s'%(p.communicate()[0]))
-                    sys.stdout.flush()
+                    # if the program exited with a non-zero exit code, don't attempt to collect performance keys
+                    status = 0
+                    if exec_status == 0:
+                        sys.stdout.write('[Executing %s/test/computePerfStats %s %s %s %s %s %s]\n'%(utildir, perfexecname, perfdir, keyfile, execlog, str(exectimeout), perfdate))
+                        sys.stdout.flush()
+                        p = py3_compat.Popen([utildir+'/test/computePerfStats',
+                                            perfexecname, perfdir, keyfile, execlog, str(exectimeout), perfdate],
+                                            stdout=subprocess.PIPE)
+                        sys.stdout.write('%s'%(p.communicate()[0]))
+                        sys.stdout.flush()
 
-                    status = p.returncode
-                    if not exectimeout and not launcher_error:
-                        if status == 0:
-                            os.unlink(execlog)
-                            sys.stdout.write('%s[Success '%(futuretest))
-                        else:
-                            sys.stdout.write('%s[Error '%(futuretest))
-                        sys.stdout.write('matching performance keys for %s/%s'%
-                                        (localdir, test_filename))
-                        if status!=0:
-                            printTestVariation(compoptsnum, compoptslist,
-                                               execoptsnum, execoptslist);
-                        sys.stdout.write(']\n')
+                        status = p.returncode
+                        if not exectimeout and not launcher_error:
+                            if status == 0:
+                                os.unlink(execlog)
+                                sys.stdout.write('%s[Success '%(futuretest))
+                            else:
+                                sys.stdout.write('%s[Error '%(futuretest))
+                            sys.stdout.write('matching performance keys for %s/%s'%
+                                            (localdir, test_filename))
+                            if status!=0:
+                                printTestVariation(compoptsnum, compoptslist,
+                                                execoptsnum, execoptslist);
+                            sys.stdout.write(']\n')
+                    else:
+                        sys.stdout.write('[Program exited with non-zero exit code — skipping perf key collection]\n')
```

Comment:
I think we want this to be flagged as an error so we can get notified when it occurs. For our testing infra, that means it should start with `[Error`. I might go with `[Error execution failed for <test_name>]\n` to match the error we report when perf tests fails to compile.